### PR TITLE
[Network] Add support to mac encryption in packet game_login

### PR DESCRIPTION
**Pull request by alisonrag**
https://github.com/OpenKore/openkore/blob/3657e3ccb79d61c4f7a379a70437bf182b0cb45d/src/Network/Send/ServerType0.pm#L199

### CODE REVIEW

File: src/Network/Send.pm [hunk 1]
```diff
@@ -407,6 +407,17 @@ sub reconstruct_game_login {
 	my ($self, $args) = @_;
 	$args->{userLevel} = 0 unless exists $args->{userLevel};
 	($args->{iAccountSID}) = $masterServer->{ip} =~ /\d+\.\d+\.\d+\.(\d+)/ unless exists $args->{iAccountSID};
+
+	if (exists $args->{mac}) {
+		my $key = pack('C16', (0x06, 0xA9, 0x21, 0x40, 0x36, 0xB8, 0xA1, 0x5B, 0x51, 0x2E, 0x03, 0xD5, 0x34, 0x12, 0x00, 0x06));
+		my $chain = pack('C16', (0x3D, 0xAF, 0xBA, 0x42, 0x9D, 0x9E, 0xB4, 0x30, 0xB4, 0x22, 0xDA, 0x80, 0x2C, 0x9F, 0xAC, 0x41));
+		my $mac = $config{mac} || "F2ADCC03771E";
```
Comment (ya4ept) [inline src/Network/Send.pm:414]: I propose to move `mac` parameter to **server.txt**

**Review by ya4ept — CHANGES_REQUESTED** [review]

File: src/Network/Send.pm (hunk 1 above)
Comment (alisonrag) [reply src/Network/Send.pm:414]: this was made to spoof the user MAC, if move to servers.txt user will have to clone the entire openkore folder to change.
here in brazil and in thailand some users use proxy so make sense use 1 mac per 1 proxy

Comment (sctnightcore) [reply src/Network/Send.pm:414]: I think use `$config{mac_address}` better ?

Comment (ya4ept) [reply src/Network/Send.pm:414]: > I think use `$config{mac_address}` better ?

+1

**Comment by ya4ept** [comment]
maybe it should be the same mac ('111111111111' and "F2ADCC03771E")?
```perl
$args->{mac} = $config{macAddress} || '111111111111' unless exists $args->{mac}; # gibberish
...
my $mac = $config{macAddress} || "F2ADCC03771E";
```

**Comment by alisonrag** [comment]
> maybe it should be the same mac ('111111111111' and "F2ADCC03771E")?
> 
> ```perl
> $args->{mac} = $config{macAddress} || '111111111111' unless exists $args->{mac}; # gibberish
> ...
> my $mac = $config{macAddress} || "F2ADCC03771E";
> ```

last time i checked the default mac address of master_login was 1111111111 in client
while in game_login it always get your right mac

**Review by ya4ept — APPROVED** [review]

**Comment by ya4ept** [comment]
@alisonrag  Do not forget: https://openkore.com/wiki/macAddress

**Comment by alisonrag** [comment]
https://openkore.com/wiki/MacAddress

**Pull request merged**